### PR TITLE
Api 33483 poa v2 signatures

### PR DESCRIPTION
**Pull request by acovrig**
## Summary
- Require rep registrationNumber

## Related issue(s)
- [API-33483](https://jira.devops.va.gov/browse/API-33483)

## Testing done
- [x] *New code is covered by unit tests*
- Postman
  - POST v2 POA validate

## What areas of the site does it impact?
v2 POA

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

# Things to know about this PR
This adds representative_id checking, meaning the test body us devs use may not work. I recommend running this from our spec against your local dev to get a working set of IDs you can use in local testing:
```
Veteran::Service::Representative.create!(representative_id: '67890', poa_codes: ['083'], first_name: 'George', last_name: 'Washington')
Veteran::Service::Organization.create!(poa: '083', name: "083 - DISABLED AMERICAN VETERANS")
```
Then you should be able to hit local with a `poa_code` of `083` and `registrationNumber` of `67890`.

### CODE REVIEW

File: modules/claims_api/spec/requests/v2/veterans/rswag_power_of_attorney_spec.rb
Comment (acovrig) [inline modules/claims_api/spec/requests/v2/veterans/rswag_power_of_attorney_spec.rb:721]: Should this be a 404?

File: modules/claims_api/spec/requests/v2/veterans/power_of_attorney_request_spec.rb
Comment (FonzMP) [reply modules/claims_api/spec/requests/v2/veterans/power_of_attorney_request_spec.rb:158]: my question as well. My understanding is we pull everything internal our system, so we should no longer require nor need the first and last name, or the signatures provided. Was curious as well when I saw the changes to 2122/valid.json above. 

```
"signatures": {
    "veteran": "",
    "representative": ""
}
```


```
- If valid registration number, include Representative's electronic signature on PDF based on first and last name found in our DB
- Include the Veteran's electronic signature on PDF based on first and last name found in MPI
```

File: modules/claims_api/app/swagger/claims_api/v1/swagger.json
Comment (FonzMP) [inline modules/claims_api/app/swagger/claims_api/v1/swagger.json:9479]: we can ignore this for this PR to cut down on LoC. Although it's not much. I also think this is in another PR so it can be resolved there.

File: modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
Comment (FonzMP) [inline modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:10088]: I think the validate docs marged this morning. May want to update this branch. @acovrig

File: modules/claims_api/spec/requests/v2/veterans/rswag_power_of_attorney_spec.rb
Comment (FonzMP) [reply modules/claims_api/spec/requests/v2/veterans/rswag_power_of_attorney_spec.rb:721]: I would have to look at what exactly is happening once we get the /validate docs merged in. it does look like this file has been changed a good bit, so we may have this resolved?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:30]: we use the constant here but not a few lines above. Wondering if we should align these and maybe create one for `INDIVIDUAL` and `ORGANIZATION` (or Rep?) to make more sense? I think we may be dependent on `FORM_NUMBER` elsewhere, but should we align these values throughout the controller where possible? 

thoughts on this as well @dellerbie ?

Comment (FonzMP) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:30]: could tack the above changes on to a ticket specific to cleaning these up if we so desire.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:51]: is this supposed to be 2122A? it's in a `validate2122` method and the shared validation gets `2122`?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:117]: do we need the ` || [] ` here? doesn't `individuals` return empty and the line above raises an error?

even if it makes it to this line, we would get an empty array and a count of 0.
```
3.2.2 :003 > [].pluck(:representative_id).uniq
 => []
3.2.2 :004 > [].pluck(:representative_id).uniq.count
 => 0
```

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb:117]: passing the `.empty?` check should mean the nil safe operators aren't needed

File: modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122/valid.json
Comment (jvcAdHoc) [inline modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122/valid.json:8]: I think an older comment asked about this. I thought we were removing the signatures param entirely for POA V2.

File: modules/claims_api/config/schemas/v2/2122.json
Comment (jvcAdHoc) [inline modules/claims_api/config/schemas/v2/2122.json:1]: the signatures attribute on line 350 should be removed

File: modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122/invalid_poa.json
Comment (jvcAdHoc) [inline modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122/invalid_poa.json:8]: You won't need signatures once it's removed from the schema

File: modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
Comment (FonzMP) [inline modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:9479]: this is interesting. two notes
1. "with that registration number" I think should be adjusted to "with registration number"
2. Invalid doesn't look good here. we could use one from the mocking like 083

**Comment by FonzMP** [comment]
is this waiting on another branch to merge before these changes go in? Or are we just waiting for the comments above to be addressed and resolved @acovrig ?

**Comment by acovrig** [comment]
> is this waiting on another branch to merge before these changes go in? Or are we just waiting for the comments above to be addressed and resolved @acovrig ?

Finishing testing and addressing those comments

File: modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb:66]: I think these changes should be in the org builder class. The signatures for 2122 and 2122a are not all in the same location.

Comment (jvcAdHoc) [reply modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb:66]: Or there should be logic to pick the org related signature attributes and or 2122a signature attributes

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/organization_controller.rb
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/organization_controller.rb:27]: can we not put this in the shared validation and return the rep at :15, for use in :20?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:46]: I'm assuming the 2122a signatures ticket will update this to work for the `representative` object?

File: modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb
Comment (jvcAdHoc) [reply modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb:66]: Yes, now that the builder is version there shouldn't be any changes in this file

File: modules/claims_api/config/schemas/v2/2122.json
Comment (jvcAdHoc) [inline modules/claims_api/config/schemas/v2/2122.json:1]: do we still need first and last name?

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:58]: I think `.first` will default to ordering by the `representative_id`. Our default has been to order by `created_at: :desc`.

File: modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:20]: Similar thing here, this might not fetch the "right" representative record

Comment (acovrig) [reply modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:20]: That should be the rep's primary_key GUID, so it should be the correct record.

Comment (jvcAdHoc) [reply modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:20]: the primary key for reps is not unique

Comment (FonzMP) [reply modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:20]: so we should move to something like the controller, correct? using the find_by for all and then checking last updated? using _that_ value?

Comment (jvcAdHoc) [reply modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:20]: Yeah, ordering a where clause by `created_at: :desc` should ensure it's the same record getting accessed. This should be an edge case. There probably aren't that many rep records with the same rep_id. However, I think for these signatures it matters to get the correct first and last name of the rep.

Comment (jvcAdHoc) [reply modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:20]: There are 117 rep_ids in staging with more than 1 record

File: modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:21]:
```suggestion
          rep = ::Veteran::Service::Representative.where(representative_id: rep_id).order(created_at: :desc).first
```

File: modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb
Comment (jvcAdHoc) [inline modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:80]: We know in V2 that the incoming `rep_id` should not be nil at this point which means we should always have the `rep` object too. I think having `rep_id = nil` in the job signature and the `unless rep_id.nil?` check could cause a confusing interpretation of not needing to pass in the rep_id. I believe the code would break here in that case as rep wouldn't be defined.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
Comment (acovrig) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:138]: It was used in the representative method, but a merge changed that at some point... I'll remove this then.

File: modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb
Comment (acovrig) [reply modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:78]: Good catch, I was originally making it an optional parameter but I guess I changed part of it when it became a required param

**Review by jvcAdHoc — DISMISSED** [review]

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
Comment (jvcAdHoc) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:139]: This might have been missed in the merge conflicts

File: modules/claims_api/config/schemas/v2/2122.json
Comment (jvcAdHoc) [inline modules/claims_api/config/schemas/v2/2122.json:11]: Was this indentation change intentional?

File: modules/claims_api/config/schemas/v2/2122.json
Comment (jvcAdHoc) [inline modules/claims_api/config/schemas/v2/2122.json:221]: I realized with removing first and last name that the form constructor needs to change. `modules/claims_api/lib/claims_api/v2/poa_pdf_constructor/organization.rb` item 16A uses the first and last name attributes

**Review by jvcAdHoc — APPROVED** [review]

**Pull request merged**
